### PR TITLE
Density error

### PR DESCRIPTION
**Pull request by chren574**

### CODE REVIEW

File: src/SPH.cpp [hunk 1]
```diff
@@ -41,45 +41,86 @@ SPH::SPH()
...
-	update_scalar_function( &m_particles.pos, m_neighbor_data, scalar_values, C_NEIGHBOR_RAD);
+	update_scalar_function(&m_particles.pos, m_neighbor_data, scalar_values);
+
+	update_kernel_values(kernel_values, &m_particles.pos, m_neighbor_data);
 
-	update_kernel_values(kernel_values, &m_particles.pos, m_neighbor_data, C_NEIGHBOR_RAD);
+	update_density_and_factors(m_particles.mass, &m_particles.pos, m_particles.dens, scalar_values, m_neighbor_data, alpha, kernel_values);
 
-	update_density_and_factors(m_particles.mass, &m_particles.pos, m_particles.dens, scalar_values, D_NR_OF_PARTICLES, m_neighbor_data, alpha, kernel_values);
+	calculate_kvi(alpha, &m_particles.vel, &m_particles.pos, m_particles.mass, m_delta_t, k_v_i, m_neighbor_data, scalar_values);
```
Comment (chren574) [inline src/SPH.cpp]: Add init_position to the constructor

File: src/SPH.cpp [hunk 2]
```diff
@@ -297,118 +397,172 @@ inline void update_kernel_values(float* kernel_values, Float3* pos, Neighbor_Dat
...
+		}
+		predicted_pressure[i].x = d_t_2 *res_x;
+		predicted_pressure[i].y = d_t_2 *res_y;
+		predicted_pressure[i].z = d_t_2 *res_z;
 	}
 }
 
-//TODO: remake this function using the predicted velocity
-void SPH::update_velocities(float dT)
+//TODO: add gradient kernal
```
Comment (Isafo) [inline src/SPH.cpp]: remove this todo

File: src/main.cpp [hunk 3]
```diff
@@ -64,7 +57,7 @@ int main() {
 	mCamera.update();
 
 	double lastTime = glfwGetTime() - 0.001f;
-	double dT = 0.0;
+	double dT = 0.f;
```
Comment (Isafo) [inline src/main.cpp]: double = float?

**Comment by Pelse862** [comment]
Bra jobbat

**Pull request merged**